runtime/src/qio/qio_formatted.c

```diff
@@ -382,8 +382,8 @@ qioerr qio_channel_read_string(const int threadsafe, const int byteorder, const
       // Figure out how many bytes are available.
       err = _peek_until_len(ch, maxlen, &peek_amt);
       num = peek_amt;
-      // Ignore EOF errors.
-      if( err && qio_err_to_int(err) == EEOF ) err = 0;
+      // Ignore EOF errors as long as we read something.
+      if( num > 0 && err && qio_err_to_int(err) == EEOF ) err = 0;
```

Comment:
If we short-circuit these conditionals, you might want to check 'num > 0' last.  I think it will be true more often than err will be, so we'd end up doing a lot more checks overall.